lib/Runtime/ByteCode/ByteCodeEmitter.cpp

```diff
@@ -6893,7 +6987,14 @@ void EmitMethodFld(ParseNode *pnode, Js::RegSlot callObjLocation, Js::PropertyId
     bool isScoped = (byteCodeGenerator->GetFlags() & fscrEval) != 0 ||
         (isRoot && callObjLocation != ByteCodeGenerator::RootObjectRegister);
 
-    EmitMethodFld(isRoot, isScoped, pnode->location, callObjLocation, propertyId, byteCodeGenerator, funcInfo, registerCacheIdForCall);
```

Comment:
Good. (I think fixing it this way may very likely save problems down the road where EmitLoadInstance is doing the wrong thing in other cases.)

--Paul

From: Taylor Woll [mailto:notifications@github.com]
Sent: Monday, March 7, 2016 5:30 PM
To: Microsoft/ChakraCore <ChakraCore@noreply.github.com>
Cc: Paul Leathers <pleath@microsoft.com>
Subject: Re: [ChakraCore] Implement interpreter support for module import/exports (#457)


In lib/Runtime/ByteCode/ByteCodeEmitter.cpp<https://github.com/Microsoft/ChakraCore/pull/457#discussion_r55303748>:

> @@ -6893,7 +6987,14 @@ void EmitMethodFld(ParseNode *pnode, Js::RegSlot callObjLocation, Js::PropertyId

>      bool isScoped = (byteCodeGenerator->GetFlags() & fscrEval) != 0 ||

>          (isRoot && callObjLocation != ByteCodeGenerator::RootObjectRegister);

>

> -    EmitMethodFld(isRoot, isScoped, pnode->location, callObjLocation, propertyId, byteCodeGenerator, funcInfo, registerCacheIdForCall);

Ok, cool. That seems to work ok.

—
Reply to this email directly or view it on GitHub<https://github.com/Microsoft/ChakraCore/pull/457/files#r55303748>.